streams/src/main/java/org/apache/kafka/streams/state/internals/InnerMeteredKeyValueStore.java

```diff
@@ -205,9 +207,9 @@ public V execute() {
             } else {
                 inner.put(typeConverter.innerKey(key), typeConverter.innerValue(value));
             }
-        } catch (final ProcessorStateException e) {
+        } catch (final StreamsException | ProducerFencedException e) {
             final String message = String.format(e.getMessage(), key, value);
```

Comment:
This isn't your code, but I'm also not too sure what's happening here. I couldn't find an example of `ProcessorStateException` for which this substitution would work.